cms/djangoapps/contentstore/views/component.py

```diff
@@ -121,119 +107,18 @@ def container_handler(request, usage_key_string):  # pylint: disable=too-many-st
-            prev_url = quote_plus(prev_url) if prev_url else None
-            next_url = quote_plus(next_url) if next_url else None
-
-            show_unit_tags = use_tagging_taxonomy_list_page()
-            unit_tags = None
-            if show_unit_tags and is_unit_page:
-                unit_tags = get_unit_tags(usage_key)
-
-            # Fetch the XBlock info for use by the container page. Note that it includes information
-            # about the block's ancestors and siblings for use by the Unit Outline.
-            xblock_info = create_xblock_info(xblock, include_ancestor_info=is_unit_page, tags=unit_tags)
-
-            if is_unit_page:
-                add_container_page_publishing_info(xblock, xblock_info)
-
-            # need to figure out where this item is in the list of children as the
-            # preview will need this
-            index = 1
-            for child in subsection.get_children():
-                if child.location == unit.location:
-                    break
-                index += 1
-
-            # Get the status of the user's clipboard so they can paste components if they have something to paste
-            user_clipboard = content_staging_api.get_user_clipboard_json(request.user.id, request)
-            library_block_types = [problem_type['component'] for problem_type in LIBRARY_BLOCK_TYPES]
-            is_library_xblock = xblock.location.block_type in library_block_types
-
-            return render_to_response('container.html', {
-                'language_code': request.LANGUAGE_CODE,
-                'context_course': course,  # Needed only for display of menus at top of page.
-                'action': action,
-                'xblock': xblock,
-                'xblock_locator': xblock.location,
-                'unit': unit,
-                'is_unit_page': is_unit_page,
-                'is_collapsible': is_library_xblock,
-                'subsection': subsection,
-                'section': section,
-                'position': index,
-                'prev_url': prev_url,
-                'next_url': next_url,
-                'new_unit_category': 'vertical',
-                'outline_url': '{url}?format=concise'.format(url=reverse_course_url('course_handler', course.id)),
-                'ancestor_xblocks': ancestor_xblocks,
-                'component_templates': component_templates,
-                'xblock_info': xblock_info,
-                'draft_preview_link': preview_lms_link,
-                'published_preview_link': lms_link,
-                'templates': CONTAINER_TEMPLATES,
-                'show_unit_tags': show_unit_tags,
-                # Status of the user's clipboard, exactly as would be returned from the "GET clipboard" REST API.
-                'user_clipboard': user_clipboard,
-                'is_fullwidth_content': is_library_xblock,
-            })
+            container_handler_context = get_container_handler_context(request, usage_key)
```

Comment:
there's a bug
You can't put statement 
```
   if is_unit_page and use_new_unit_page(course.id):
                return redirect(get_unit_url(course.id, unit.location))
```
Inside `get_container_handler_context` function.
Because `redirect()` returns Response object, and should be called within the View itself.

Currently on dev server the following error is thrown when enabling feature flag [contentstore.new_studio_mfe.use_new_unit_page](https://cms-studio-mfe-dev.raccoongang.com/admin/waffle/flag/23/change/)

```
│[2024-01-06 13:33:38 +0000] [47665] [INFO] Autorestarting worker after current request.                                                                                                                                                                                                                                                                                                                                            │
│2024-01-06 13:33:38,471 ERROR 47665 [root] [user None] [ip None] signals.py:22 - Uncaught exception from None                                                                                                                                                                                                                                                                                                                      │
│Traceback (most recent call last):                                                                                                                                                                                                                                                                                                                                                                                                 │
│  File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/django/core/handlers/exception.py", line 47, in inner                                                                                                                                                                                                                                                                                                             │
│    response = get_response(request)                                                                                                                                                                                                                                                                                                                                                                                               │
│  File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/django/core/handlers/base.py", line 181, in _get_response                                                                                                                                                                                                                                                                                                         │
│    response = wrapped_callback(request, *callback_args, **callback_kwargs)                                                                                                                                                                                                                                                                                                                                                        │
│  File "/usr/lib/python3.8/contextlib.py", line 75, in inner                                                                                                                                                                                                                                                                                                                                                                       │
│    return func(*args, **kwds)                                                                                                                                                                                                                                                                                                                                                                                                     │
│  File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/django/views/decorators/http.py", line 40, in inner                                                                                                                                                                                                                                                                                                               │
│    return func(request, *args, **kwargs)                                                                                                                                                                                                                                                                                                                                                                                          │
│  File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/django/contrib/auth/decorators.py", line 21, in _wrapped_view                                                                                                                                                                                                                                                                                                     │
│    return view_func(request, *args, **kwargs)                                                                                                                                                                                                                                                                                                                                                                                     │
│  File "/edx/app/edxapp/edx-platform/cms/djangoapps/contentstore/views/component.py", line 121, in container_handler                                                                                                                                                                                                                                                                                                               │
│    return render_to_response('container.html', container_handler_context)                                                                                                                                                                                                                                                                                                                                                         │
│  File "/edx/app/edxapp/edx-platform/common/djangoapps/edxmako/shortcuts.py", line 188, in render_to_response                                                                                                                                                                                                                                                                                                                      │
│    return HttpResponse(render_to_string(template_name, dictionary, namespace, request), **kwargs)                                                                                                                                                                                                                                                                                                                                 │
│  File "/edx/app/edxapp/edx-platform/common/djangoapps/edxmako/shortcuts.py", line 178, in render_to_string                                                                                                                                                                                                                                                                                                                        │
│    return template.render(dictionary, request)                                                                                                                                                                                                                                                                                                                                                                                    │
│  File "/edx/app/edxapp/edx-platform/common/djangoapps/edxmako/template.py", line 77, in render                                                                                                                                                                                                                                                                                                                                    │
│    context_dictionary.update(context)                                                                                                                                                                                                                                                                                                                                                                                             │
│ValueError: dictionary update sequence element #0 has length 0; 2 is required                                                                                                                                                                                                                                                                                                                                                      │
│2024-01-06 13:33:38,491 ERROR 47665 [django.request] [user 19] [ip 88.135.201.166] log.py:224 - Internal Server Error: /container/block-v1:eDx+123+12+type@vertical+block@bf197d5f929d421980c6acbd8583964d                                                                                                                                                                                                                         │
│Traceback (most recent call last):                                                                                                                                                                                                                                                                                                                                                                                                 │
│  File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/django/core/handlers/exception.py", line 47, in inner                                                                                                                                                                                                                                                                                                             │
│    response = get_response(request)                                                                                                                                                                                                                                                                                                                                                                                               │
│  File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/django/core/handlers/base.py", line 181, in _get_response                                                                                                                                                                                                                                                                                                         │
│    response = wrapped_callback(request, *callback_args, **callback_kwargs)                                                                                                                                                                                                                                                                                                                                                        │
│  File "/usr/lib/python3.8/contextlib.py", line 75, in inner                                                                                                                                                                                                                                                                                                                                                                       │
│    return func(*args, **kwds)                                                                                                                                                                                                                                                                                                                                                                                                     │
│  File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/django/views/decorators/http.py", line 40, in inner                                                                                                                                                                                                                                                                                                               │
│    return func(request, *args, **kwargs)                                                                                                                                                                                                                                                                                                                                                                                          │
│  File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/django/contrib/auth/decorators.py", line 21, in _wrapped_view                                                                                                                                                                                                                                                                                                     │
│    return view_func(request, *args, **kwargs)                                                                                                                                                                                                                                                                                                                                                                                     │
│  File "/edx/app/edxapp/edx-platform/cms/djangoapps/contentstore/views/component.py", line 121, in container_handler                                                                                                                                                                                                                                                                                                               │
│    return render_to_response('container.html', container_handler_context)                                                                                                                                                                                                                                                                                                                                                         │
│  File "/edx/app/edxapp/edx-platform/common/djangoapps/edxmako/shortcuts.py", line 188, in render_to_response                                                                                                                                                                                                                                                                                                                      │
│    return HttpResponse(render_to_string(template_name, dictionary, namespace, request), **kwargs)                                                                                                                                                                                                                                                                                                                                 │
│  File "/edx/app/edxapp/edx-platform/common/djangoapps/edxmako/shortcuts.py", line 178, in render_to_string                                                                                                                                                                                                                                                                                                                        │
│    return template.render(dictionary, request)                                                                                                                                                                                                                                                                                                                                                                                    │
│  File "/edx/app/edxapp/edx-platform/common/djangoapps/edxmako/template.py", line 77, in render                                                                                                                                                                                                                                                                                                                                    │
│    context_dictionary.update(context)                                                                                                                                                                                                                                                                                                                                                                                             │
│ValueError: dictionary update sequence element #0 has length 0; 2 is required                                                                                                                                                                                                                                                                                                                                                      │
│[2024-01-06 13:33:38 +0000] [47665] [INFO] Worker exiting (pid: 47665)                                                                                                                                                                                                                                                                                                                                                             │
│[2024-01-06 13:33:41 +0000] [62159] [INFO] Booting worker with pid: 62159 
```

Ticket returned to rework

Reply:
fixed here https://github.com/raccoongang/edx-platform/pull/2487